File: library/db/src/main/java/org/keynote/godtools/android/db/LanguageMapper.java
package org.keynote.godtools.android.db;

import android.content.ContentValues;
import android.database.Cursor;
import android.support.annotation.NonNull;

import org.cru.godtools.model.Language;

import static org.keynote.godtools.android.db.Contract.LanguageTable.COLUMN_ADDED;
import static org.keynote.godtools.android.db.Contract.LanguageTable.COLUMN_CODE;
import static org.keynote.godtools.android.db.Contract.LanguageTable.COLUMN_NAME;

final class LanguageMapper extends BaseMapper<Language> {
    @Override
    protected void mapField(@NonNull final ContentValues values, @NonNull final String field,
                            @NonNull final Language language) {
        switch (field) {
            case COLUMN_CODE:
                values.put(field, serialize(language.getCode()));
                break;
            case COLUMN_ADDED:
                values.put(field, language.isAdded());
                break;
            case COLUMN_NAME:
                values.put(field, language.getLanguageName());
                break;
            default:
                super.mapField(values, field, language);
                break;
        }
    }

    @NonNull
    @Override
    protected Language newObject(@NonNull final Cursor c) {
        return new Language();
    }

    @NonNull
    @Override
    public Language toObject(@NonNull final Cursor c) {
        final Language language = super.toObject(c);

        language.setCode(getLocale(c, COLUMN_CODE, Language.INVALID_CODE));
        language.setAdded(getBool(c, COLUMN_ADDED, false));
        language.setLanguageName(getString(c, COLUMN_NAME, null));

        return language;
    }
}

File: library/db/src/main/java/org/keynote/godtools/android/db/LanguageMapper.kt
package org.keynote.godtools.android.db

import android.content.ContentValues
import android.database.Cursor
import org.ccci.gto.android.common.util.database.getLocale
import org.ccci.gto.android.common.util.database.getString
import org.cru.godtools.model.Language
import org.keynote.godtools.android.db.Contract.LanguageTable.COLUMN_ADDED
import org.keynote.godtools.android.db.Contract.LanguageTable.COLUMN_CODE
import org.keynote.godtools.android.db.Contract.LanguageTable.COLUMN_NAME

internal object LanguageMapper : BaseMapper<Language>() {
    override fun mapField(values: ContentValues, field: String, obj: Language) {
        when (field) {
            COLUMN_CODE -> values.put(field, serialize(obj.code))
            COLUMN_NAME -> values.put(field, obj.name)
            COLUMN_ADDED -> values.put(field, obj.isAdded)
            else -> super.mapField(values, field, obj)
        }
    }

    override fun newObject(c: Cursor) = Language()
    override fun toObject(c: Cursor) = super.toObject(c).apply {
        code = c.getLocale(COLUMN_CODE, Language.INVALID_CODE)
        name = c.getString(COLUMN_NAME)
        isAdded = getBool(c, COLUMN_ADDED, false)
    }
}
